opentelemetry/src/baggage.rs

```diff
@@ -146,10 +149,41 @@ impl Baggage {
         S: Into<BaggageMetadata>,
     {
         let (key, value, metadata) = (key.into(), value.into(), metadata.into());
-        if self.insertable(&key, &value, &metadata) {
-            self.inner.insert(key, (value, metadata))
-        } else {
-            None
+        if !key.as_str().is_ascii() {
+            return None;
+        }
```

Comment:
I understand this is the existing code, but baggage specs only allow for the printable ASCII characters. Probably, good to also check for control characters.

Reply:
I'm not sure how to fix that in efficient way, so I would prefer to leave this to someone more competent.